components/webln.js

```diff
@@ -0,0 +1,59 @@
+import { createContext, useContext, useEffect, useState } from 'react'
+
+const WebLNContext = createContext({})
+
+export function WebLNProvider ({ children }) {
+  const [provider, setProvider] = useState(null)
+  const [info, setInfo] = useState(null)
+  // NOTE these functions are undefined initially - can this be a problem?
+  const [launchModal, setLaunchModal] = useState()
+  const [launchPaymentModal, setLaunchPaymentModal] = useState()
+  const [closeModal, setCloseModal] = useState()
+  const [isConnected, setIsConnected] = useState()
+
+  useEffect(() => {
+    const unsub = []
+    async function effect () {
+      const [isConnected, onConnected, onDisconnected, requestProvider, launchModal, launchPaymentModal, closeModal] = await import('@getalby/bitcoin-connect-react').then(
+        (mod) => [mod.isConnected, mod.onConnected, mod.onDisconnected, mod.requestProvider, mod.launchModal, mod.launchPaymentModal, mod.closeModal]
+      )
+
+      // if you want to store a function, you need to wrap it with another function because of updater functions
+      // see https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state
+      setLaunchModal(() => launchModal)
```

Comment:
These functions are imported and then passed as state, could they just be imported where they are actually used? actually I don't think they are even used yet, are they?